From 0e408d461a9fd9e7f48136083a472a6836cdad1f Mon Sep 17 00:00:00 2001
From: Sahil Takiar <takiar.sahil@gmail.com>
Date: Tue, 14 Mar 2017 08:15:39 -0700
Subject: [PATCH 1041/1164] CDH-51717 HIVE-16192 : QTestUtil doesn't correctly
 set the last command when running "test" commands
 (Sahil Takiar via Ashutosh Chauhan)

Signed-off-by: Ashutosh Chauhan <hashutosh@apache.org>
(cherry picked from commit b6e39c9e87ba275f5d243f83a2231eeb8f991890)

Change-Id: I33d0d954ab4dbd49fc1e26737ae8e9b0d956afef
---
 .../java/org/apache/hadoop/hive/ql/QTestUtil.java  |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/itests/util/src/main/java/org/apache/hadoop/hive/ql/QTestUtil.java b/itests/util/src/main/java/org/apache/hadoop/hive/ql/QTestUtil.java
index 0d78986..3785957 100644
--- a/itests/util/src/main/java/org/apache/hadoop/hive/ql/QTestUtil.java
+++ b/itests/util/src/main/java/org/apache/hadoop/hive/ql/QTestUtil.java
@@ -1314,6 +1314,10 @@ private int executeTestCommand(final String command) {
     commandArgs = commandArgs.replaceAll("\\$\\{hiveconf:hive\\.metastore\\.warehouse\\.dir\\}",
       wareHouseDir);
 
+    if (SessionState.get() != null) {
+      SessionState.get().setLastCommand(commandName + " " + commandArgs.trim());
+    }
+
     enableTestOnlyCmd(SessionState.get().getConf());
 
     try {
@@ -2206,7 +2210,7 @@ public boolean accept(File dir, String name) {
   public void failed(int ecode, String fname, String debugHint) {
     String command = SessionState.get() != null ? SessionState.get().getLastCommand() : null;
     String message = "Client execution failed with error code = " + ecode +
-        (command != null ? " running " + command : "") + "fname=" + fname +
+        (command != null ? " running \"" + command : "") + "\" fname=" + fname + " " +
         (debugHint != null ? debugHint : "");
     LOG.error(message);
     Assert.fail(message);
-- 
1.7.9.5

